tracer/src/Datadog.Trace/ExtensionMethods/SpanExtensions.cs

```diff
@@ -26,16 +27,27 @@ public static class SpanExtensions
         /// <param name="samplingPriority">The new sampling priority for the trace.</param>
         public static void SetTraceSamplingPriority(this ISpan span, SamplingPriority samplingPriority)
```

Comment:
It's not very clear to me why in that case we set a `Manual` sampling mechanism. Should we rename the extension to make it clearer. The other extension method as the same name but a different behaviour which I find hard to follow

```suggestion
        public static void SetManualTraceSamplingPriority(this ISpan span, SamplingPriority samplingPriority)
```

Reply:
This public extension method already existed and from now on should only be used by users to set the sampling priority (so it's always `Manual`).

I used to have some comments here and I though I removed the overload, but looks like I lost that with all the revisions. I'll clean this up, thanks.

Reply:
I confirmed that neither extension method is used internally by Datadog. I removed the `internal` overload and added a comment to this `public` one. I hope we can deprecate it soon and remove it in tracer 3.0 along with the `SamplingPriority` enum.